code/src/uarm_swift_pro.cpp

```diff
@@ -0,0 +1,46 @@
+#include <uarm_swift_pro.hpp>
+
+namespace r2d2::robot_arm {
+    uarm_swift_pro_c::uarm_swift_pro_c(const r2d2::hardware_usart_c &usart_bus)
+        : usart_bus(usart_bus) {
+    }
+    uarm_swift_pro_c::uarm_swift_pro_c(unsigned int &bautrate,
+                                       r2d2::uart_ports_c &usart_port)
+        : usart_bus(bautrate, usart_port) {
+    }
+
+    void uarm_swift_pro_c::move_head_to_coordinate(const vector3i_c &coordinate,
+                                                   const uint16_t &speed) {
+        gcode_generator.coordinate_to_gcode(coordinate, speed);
+        this->send_command(gcode_generator.get_buffer());
+    }
+
+    void
+    uarm_swift_pro_c::move_head_to_coordinate(const vector3i_c &coordinate) {
+        gcode_generator.coordinate_to_gcode(coordinate, default_speed);
+        this->send_command(gcode_generator.get_buffer());
+    }
+
+    void uarm_swift_pro_c::rotate_head(const int16_t &rotation) {
+        // TODO impl rotate head uarm swift pro
+    }
+
+    void uarm_swift_pro_c::init() {
+        this->send_command("#n M2122 V1\n"); // report when stop
+    }
+
+    bool uarm_swift_pro_c::check_connection() {
+        // TODO impl check_connection()
+        return true;
+    }
+
+    bool uarm_swift_pro_c::send_command(const char *command) {
+        this->usart_bus << command;
+        return true;
```

Comment:
This always returns true? Is this really needed if it is always true?

Reply:
For now, it will be true, in later implementations, we will check if the command was successful. When the uarm is sent gcode with `#1` at the start it will return `$1 ok` when the command was successfully executed